lib/rubocop/cop/rspec/capybara/specific_matcher.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Capybara
+        # Checks for there is a more specific matcher offered by Capybara.
+        #
+        # @example
+        #
+        #   # bad
+        #   expect(page).to have_selector('button')
+        #   expect(page).to have_none_of_selectors('button.cls')
+        #   expect(page).to have_css('button')
+        #   expect(page).to have_no_css('a.cls', exact_text: 'foo')
+        #   expect(page).to have_css('table.cls')
+        #   expect(page).to have_css('select')
+        #
+        #   # good
+        #   expect(page).to have_button
+        #   expect(page).to have_button(class: 'cls')
+        #   expect(page).to have_button
+        #   expect(page).to have_no_link('foo', class: 'cls')
+        #   expect(page).to have_table(class: 'cls')
+        #   expect(page).to have_select
+        #
+        class SpecificMatcher < Base
+          MSG = 'Use `%<good_matcher>s`.'
+          RESTRICT_ON_SEND = %i[have_selector have_css have_no_css].freeze
+          SPECIFIC_MATCHER = {
+            button: 'button',
+            a: 'link',
+            table: 'table',
+            select: 'select'
+          }.freeze
+
+          # @!method first_argument(node)
+          def_node_search :first_argument, <<-PATTERN
+            (send nil? _ (str $_) ?... )
+          PATTERN
+
+          def on_send(node)
+            return unless node.first_argument.str_type?
```

Comment:
This seems redundant now - the node patter would only match for a string.

Reply:
Indeed is it, I'll fix it.